src/components/PrivateRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Redirect, Route, RouteProps } from 'react-router-dom';
+
+import { RootStore } from '../redux/reducers/rootReducer';
+
+interface PrivateRouteProps extends RouteProps {
+  component: React.ComponentType<any>;
+  isAuthorized?: boolean;
+}
+
+const PrivateRoute: React.FC<PrivateRouteProps> = ({ component: Component, isAuthorized, ...rest }) => {
+  const auth = useSelector((store: RootStore) => store.auth);
+  const autorized = isAuthorized ? isAuthorized : auth.isLoggedIn;
```

Comment:
Hvor kommer `isAuthorized` fra? Ser ikke at den blir sendt som prop noe sted

Reply:
tanken var at det er flere måter å sette isAuthorized, så den er optional. Når vi får roller etterhvert, så vil det være noen ruter som kun er tilgjengelig for kurator-rollen f.eks. men kan evt fjerne den nå også får vi legge den til igjen når det blir aktuelt?